build.sh

```diff
@@ -123,18 +125,17 @@ COMMIT_SHA=$( git rev-parse HEAD )
 echo "PIPELINE: $PIPELINE"
 
 # Display system info
-hostname
-whoami
-cat /etc/*release
-lscpu
-free
-df -h || true
-pwd
-docker info
-docker version
-
-# Terminate the build on errors
-set -e
+(
+    hostname
+    whoami
+    cat /etc/*release
+    lscpu
+    free
+    df -h
+    pwd
+    docker info
+    docker version
+) || true
```

Comment:
seems like a hack. `set -e` could begin from after this section of commands.

Reply:
presently `set -e` is already below this section. so maybe add `set -u` here: https://github.com/startersclan/docker-sourceservers/pull/126#discussion_r1130976055

Reply:
that's why `set -e` often limiting, forcing the script to be written around terminating errors

Reply:
yes, but in this case outputting these info requires tools that some distros wont have anyway.

Reply:
again i don't think it's good to default to `set -e` for the entire script. `set -e` should like presently begin at the build steps and not apply at begin/validation steps

Reply:
it's really a matter of preference / style, but in general i prefer using `set -e` because it makes every line of the script an assertion. imo, `set -e the script unexpectedly terminating with `set -e` is better than not using `set -e` and having the script fail much further down the actual line that exited with non-`0`.

Reply:
it's really a matter of preference / style, but in general i prefer using `set -e` because it makes every line of the script an assertion. imo, the script unexpectedly terminating with `set -e` is better than not using `set -e` and having the script fail much further down the actual line that exited with non-`0`.

Reply:
never thought this script should be used on git for windows, but it will work with windows with this fix.

Reply:
no let's just leave `set -e` where it presently is. open another with `set -u` at the top is fine

Reply:
well, then everything before the `set -e` may exit with non-`0`, and not just this section.

Reply:
must make sure error handling is fine before `set -e`

close ok

Reply:
> must make sure error handling is fine before set -e

which is why `set -e` at top of script is better

Reply:
not the same thing

Reply:
actually, the `set -e` carries over into the subshell, so the first failure of any command terminate the subshell. hence, `set +e` needs to be set inside the subshell.